sale_require_po_doc/models/sale_order.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    customer_need_po = fields.Boolean(
+        related="partner_id.customer_need_po",
+        string="Customer Requires PO",
+        readonly=True,
+    )
+    sale_doc = fields.Text(
+        related="partner_id.sale_doc",
+        string="Sales Require Documentation",
+        readonly=True,
+    )
+    sale_document_option = fields.Selection(
+        [
+            ("no_need", "No documentation needed for this Sale"),
+            ("done", "Documentation required obtained and archived"),
+        ],
+        string="Sale Documentation",
+    )
+    sale_document_note = fields.Text(string="Sale Documentation Notes")
+
+    def action_confirm(self):
+        for order in self:
+            if order.customer_need_po and not order.client_order_ref:
```

Comment:
IMHO the need po is not well named as this is for Customer Reference (I know the reference is (maybe) a PO on customer side - but we are in our side context).

See https://github.com/odoo/odoo/blob/14.0/addons/sale/models/sale.py#L161